src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import {
+  Container,
+  TitleContainer,
+  Title,
+  CloseButton,
+  Content,
+  Fade,
+} from './Modal.style';
+
+export interface ModalProps {
+  isOpen: boolean;
+  hide: () => void;
+  modalContent: JSX.Element;
```

Comment:
with the changes I suggested to update this to `children`, I believe you can do `children: ReactNode`